### PR TITLE
Fix/Upgrade nvd api v1 to v2

### PR DESCRIPTION
**Pull request by oussamaessaji**
The NVD endpoint at https://services.nvd.nist.gov/rest/json/cve/1.0/ has been [deprecated](https://nvd.nist.gov/general/news/change-timeline). This deprecation causes the agent to fail silently, as we catch the exceptions and default to a defined CVE. 
The unit test did not catch the exception because we used to mock the requests. To address this issue, follow these steps:

- Switch to the new endpoint  "https://services.nvd.nist.gov/rest/json/cves/2.0?cveId=".
- Pass the NVD API key, as it enforces a rate limit of 50 requests per 30 seconds, compared to the normal one without an API key, which has a rate limit of 5 requests per 30 seconds.
- Remove the mock from the unit test to test the API.

### CODE REVIEW

File: agent/cve_service_api.py [hunk 1]
```diff
@@ -31,49 +31,55 @@ class CVE:
...
     """Given a CVE ID, retrieve the risk rating from the MITRE CVE API.
     Args:
         cve_id: the cve id to obtain its details
     Returns:
         CVE object.
     """
     url = f"{CVE_MITRE_BASE_URL}{cve_id}"
-
-    response = requests.get(url, timeout=REQUEST_TIMEOUT)
+    # api_key = "08f18e8b-7a76-4398-b90d-33e0ce93b442"
```
Comment (PiranhaSa) [inline agent/cve_service_api.py:42]: not a good idea to have an api key hard coded

File: tests/cve_service_api_test.py [hunk 2]
```diff
@@ -7,105 +7,17 @@
...
-            "lastModifiedDate":"2023-04-03T20:15Z"
-         }
-      ]
-   }
-}
-"""
-    requests_mock.get(
-        re.compile("https://services.nvd.nist.gov/.*"),
-        text=cve_data_json,
+def testGetCveData_withResponse_returnRiskRating() -> None:
```
Comment (PiranhaSa) [inline tests/cve_service_api_test.py:10]: docString please

File: tests/osv_wrapper_test.py [hunk 3]
```diff
@@ -37,7 +37,10 @@ def testParseResults_withValidFile_returnData(fake_osv_output: str) -> None:
         "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-1941"
         in parsed_data_list[0].technical_detail
     )
-    assert "version `3.18.3`" in parsed_data_list[0].technical_detail
+    assert (
+        "We recommend upgrading to versions 3.18.3, 3.19.5, 3.20.2, 3.21.6"
+        in parsed_data_list[0].technical_detail
+    )
```
Comment (PiranhaSa) [inline tests/osv_wrapper_test.py:43]: why you don't combine these two ?

File: agent/cve_service_api.py (hunk 1 above)
Comment (oussamaessaji) [reply agent/cve_service_api.py]: Revoked and applied!

**Review by 3asm — APPROVED** [review]

**Pull request merged**